packages/grid/src/Grid.tsx

```diff
@@ -1,81 +1,313 @@
-      const newRows = selectedRows.filter((el) => row !== el);
-      changeSelectedRows(newRows);
-    } else {
-      changeSelectedRows([...selectedRows, row]);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  const onChangeExpand = useCallback(
+    (index: number, childrens: IItem[]) => {
+      if (mappedRows[index].isExpand) {
+        console.log('сработает если открыто');
+        let childrensLength = 0;
+        for (let i = index + 1; i < mappedRows.length; i++) {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          if (mappedRows[i].expandLevel) childrensLength += 1;
+          else break;
+        }
+        const newMappedRows = [
+          ...mappedRows.slice(0, index + 1),
+          ...mappedRows.slice(index + 1 + childrensLength),
+        ];
+        const withNewExpand = setIn(newMappedRows, false, [
+          String(index),
+          'isExpand',
+        ]);
+        changeMappedRows(withNewExpand);
+      } else {
+        const parentExpandLevel = mappedRows[index].expandLevel || 0;
+        const newChildrens = childrens.map(
+          (el: IItem): IMappedItem => ({
+            ...el,
+            expandLevel: parentExpandLevel + 1,
+            key: guid(),
+          }),
+        );
+        const newMappedRows = [
+          ...mappedRows.slice(0, index + 1),
+          ...newChildrens,
+          ...mappedRows.slice(index + 1),
+        ];
+        const withNewExpand = setIn(newMappedRows, true, [String(index), 'isExpand']);
+        changeMappedRows(withNewExpand);
+      }
+    },
+    [mappedRows],
+  );
+
+  const handleSelect = useCallback(
+    (e, key) => {
+      if (isDisableSelect
+           || e.target.tagName === 'svg'
+           || e.target.tagName === 'path'
+           || e.target.tagName === 'BUTTON'
+      ) return;
+      if (isMultiSelect) {
+        const newSelectedRows = addOrDeleteItemFromArray(selectedRows, key);
+        onSelect(
+          deletePropsFromObjects(
+            mappedRows.filter((el: IMappedItem) => newSelectedRows.some((id) => id === el.key)), 'key', 'expandLevel',
+          ),
+        );
+        changeSelectedRows(newSelectedRows);
+        return;
+      }
+      if (selectedRows[0] === key) {
+        onSelect({});
+        changeSelectedRows([]);
+      } else {
+        const selectedRow = {
+          ...mappedRows
+            .find((el: IMappedItem) => el.key === key),
+        } as IMappedItem;
+        delete selectedRow.key;
+        delete selectedRow.expandLevel;
+        onSelect(selectedRow);
+        changeSelectedRows([key]);
+      }
+    },
+    [isDisableSelect, isMultiSelect, selectedRows, onSelect, mappedRows],
+  );
+
+  const cell = ({
+    columnIndex, key, parent, rowIndex, style,
+  }: GridCellProps) => {
+    const content = mappedRows[rowIndex][mappedColumns[columnIndex].field];
+    const expandLevel = (!columnIndex && mappedRows[rowIndex].expandLevel) || 0;
```

Comment:
!columnIndex could you explain why this is necessary.

Reply:
expand offset only at first column